### PR TITLE
EN-16378 Handle grouping derived views.

### PR DESCRIPTION
**Pull request by chitang**

### CODE REVIEW

File: common-pg/src/main/scala/com/socrata/pg/soql/ExprSqlizer.scala [hunk 1]
```diff
@@ -125,7 +125,7 @@ object ColumnRefSqlizer extends Sqlizer[ColumnRef[UserColumnId, SoQLType]] {
           escape: Escape): ParametricSql = {
     val tableMap = ctx(TableAliasMap).asInstanceOf[Map[String, String]]
     reps.get(QualifiedUserColumnId(expr.qualifier, expr.column)) match {
-      case Some(rep) if ctx(InnermostSoql) == true => // scalastyle:off simplify.boolean.expression
+      case Some(rep) if ctx(InnermostSoql) == true || expr.qualifier.nonEmpty => // scalastyle:off simplify.boolean.expression
```
Comment (aerust) [inline common-pg/src/main/scala/com/socrata/pg/soql/ExprSqlizer.scala]: Is there a reason `ctx(InnermostSoql) == true` is being used here?

**Comment by aerust** [comment]
Can you explain in comment or commit message why adding `|| expr.qualifier.nonEmpty` is the right thing to do? (both for my sake and for future sake)

File: common-pg/src/main/scala/com/socrata/pg/soql/ExprSqlizer.scala (hunk 1 above)
Comment (chitang) [inline common-pg/src/main/scala/com/socrata/pg/soql/ExprSqlizer.scala]: not inner most is handled in the next case statement.  Until joins, innermost is the only one that directly interacts with t tables.

Comment (chitang) [inline common-pg/src/main/scala/com/socrata/pg/soql/ExprSqlizer.scala]: i am adding tests related to this.

Comment (aerust) [inline common-pg/src/main/scala/com/socrata/pg/soql/ExprSqlizer.scala]: oh. more of a question of why `== true`

**Comment by aerust** [comment]
Okay, once it builds, looks good to me. :sheep:

**Pull request merged**